Sources/Extensions/SpriteKit/SKNodeExtensions.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  SpriteKitExtensions.swift
+//  SwifterSwift
+//
+//  Created by Olivia Brown on 5/28/18.
+//  Copyright © 2018 SwifterSwift
+//
+
+import SpriteKit
+
+// MARK: - Methods
+public extension SKNode {
+    /// SwifterSwift: Return an array of all SKNode descendants
+    ///
+    ///         mySKNode.descendants() -> [childNodeOne, childNodeTwo]
+    ///
+    public func descendants() -> [SKNode] {
+        let firstLevelDescendants = children.compactMap { $0 }
```

Comment:
What's the need for `compactMap`? I don't think it's useful here since the values of children are non-nil.

Reply:
You can also place this whole function into a single line:
```Swift
return children + children.reduce(into: [SKNode]()) { $0 += $1.descendants() }
```

Reply:
It may not matter too much, but I just want to comment that it's not a standard search path (as far as I'm aware) that's been implemented here. It's neither depth-first nor breadth-first, but rather a kind of hybrid. I guess it can be forgiven if you make it a one-liner 😛

Reply:
Good catch on compactMap. Definitely a lot cleaner!

Reply:
Ok @guykogus I see what you're saying here. I guess order _technically_ has meaning here cuz Array is an ordered collection. But I'd be fine with just getting the ball rolling. If someone wants to come back later and implement a standard search path then I'm open to a PR 👍